hadoop-hdds/rocks-native/src/main/patches/rocks-native.patch

```diff
@@ -16,592 +16,557 @@
++
++
++
++extern const uint64_t kBlockBasedTableMagicNumber;
++extern const uint64_t kLegacyBlockBasedTableMagicNumber;
++extern const uint64_t kPlainTableMagicNumber;
++extern const uint64_t kLegacyPlainTableMagicNumber;
++
++Status RawSstFileReader::GetTableReader(const std::string& file_path) {
++  // Warning about 'magic_number' being uninitialized shows up only in UBsan
++  // builds. Though access is guarded by 's.ok()' checks, fix the issue to
++  // avoid any warnings.
++  uint64_t magic_number = Footer::kNullTableMagicNumber;
++
++  // read table magic number
++  Footer footer;
++
++  const auto& fs = options_.env->GetFileSystem();
++  std::unique_ptr<FSRandomAccessFile> file;
++  uint64_t file_size = 0;
++  FileOptions fopts = rep_->soptions_;
++  fopts.temperature = file_temp_;
++  Status s = fs->NewRandomAccessFile(file_path, fopts, &file, nullptr);
++  if (s.ok()) {
++    s = fs->GetFileSize(file_path, IOOptions(), &file_size, nullptr);
++  }
++
++  // check empty file
++  // if true, skip further processing of this file
++  if (file_size == 0) {
++    return Status::Aborted(file_path, "Empty file");
++  }
++
++  rep_->file_.reset(new RandomAccessFileReader(std::move(file), file_path));
++
++  FilePrefetchBuffer prefetch_buffer(
++      0 /* readahead_size */, 0 /* max_readahead_size */, true /* enable */,
++      false /* track_min_offset */);
++  if (s.ok()) {
++    const uint64_t kSstDumpTailPrefetchSize = 512 * 1024;
++    uint64_t prefetch_size = (file_size > kSstDumpTailPrefetchSize)
++                                 ? kSstDumpTailPrefetchSize
++                                 : file_size;
++    uint64_t prefetch_off = file_size - prefetch_size;
++    IOOptions opts;
++    s = prefetch_buffer.Prefetch(opts, rep_->file_.get(), prefetch_off,
++                                 static_cast<size_t>(prefetch_size),
++                                 Env::IO_TOTAL /* rate_limiter_priority */);
++
++    s = ReadFooterFromFile(opts, rep_->file_.get(), &prefetch_buffer, file_size,
++                           &footer);
++  }
++  if (s.ok()) {
++    magic_number = footer.table_magic_number();
++  }
++
++  if (s.ok()) {
++    if (magic_number == kPlainTableMagicNumber ||
++        magic_number == kLegacyPlainTableMagicNumber) {
++      rep_->soptions_.use_mmap_reads = true;
++
++      fs->NewRandomAccessFile(file_path, fopts, &file, nullptr);
++      rep_->file_.reset(new RandomAccessFileReader(std::move(file), file_path));
++    }
++
++    s = ROCKSDB_NAMESPACE::ReadTableProperties(
++        rep_->file_.get(), file_size, magic_number, rep_->ioptions_, &(rep_->table_properties_),
++        /* memory_allocator= */ nullptr, (magic_number == kBlockBasedTableMagicNumber)
++                                             ? &prefetch_buffer
++                                             : nullptr);
++    if (!s.ok()) {
++      if (!silent_) {
++        fprintf(stderr, "Not able to read table properties\n");
++      }
```

Comment:
This can be moved inside else block on line 311-313.

Reply:
nit: this is a simple logging if I'm not wrong. And I guess you added it for debugging.
I don't see this in [sst_file_dumper](https://github.com/facebook/rocksdb/blob/master/table/sst_file_dumper.cc#L73).

Please either remove this or move it to else.

Reply:
the value for s is set just a line above.

Reply:
You have a if-else from line 296-313, just after this. There is no change in value of `s` in between. I don't see any issue in moving this inside else block on lines 311-313.
```
+      if (!silent_) {
+        fprintf(stderr, "Not able to read table properties\n");
+      }
```
What am I missing here?

Reply:
oh I missed it. I have fixed this issue.